apps/web-mzima-client/src/app/post/location-select/location-select.component.ts

```diff
@@ -87,22 +100,61 @@ export class LocationSelectComponent implements OnInit, ControlValueAccessor {
     this.mapReady = true;
   }
 
-  onMapReady(map: Map) {
-    this.map = map;
-    control.zoom({ position: 'bottomleft' }).addTo(map);
-    this.map.panTo(this.location);
-    this.addMarker();
+  ngAfterViewInit() {
+    // change tracking for search when entering text in geocoder search input (and debounce to reduce geocoding requests sent)
+    const geocoderInputElement = this.geocoderControl.getContainer().querySelector('input');
+    this.geocoderSubscription = fromEvent(geocoderInputElement, 'input')
+      .pipe(
+        filter(Boolean),
+        debounceTime(600),
+        distinctUntilChanged(),
+        tap(() => {
+          this.geocoderControl.options.placeholder = geocoderInputElement.value;
+          this.geocoderControl._input.value = geocoderInputElement.value;
+          this.geocoderControl._geocode();
+        }),
+      )
+      .subscribe();
+  }
+
+  ngOnDestroy() {
+    this.geocoderSubscription?.unsubscribe();
+  }
```

Comment:
please use **untilDestroyed(this)**
```
fromEvent(geocoderInputElement, 'input')
  .pipe(
    filter(Boolean),
    debounceTime(600),
    distinctUntilChanged(),
    tap(() => {
      this.geocoderControl.options.placeholder = geocoderInputElement.value;
      this.geocoderControl._input.value = geocoderInputElement.value;
      this.geocoderControl._geocode();
    }),
    untilDestroyed(this),
  )
  .subscribe();
```

and remove 
```
  ngOnDestroy() {
    this.geocoderSubscription?.unsubscribe();
  }
```

Reply:
Ok @kvengerov